### PR TITLE
Create GitHub teams from CSV

### LINKED ISSUE #194: CSV team creation (see issue details for format)
### CSV Format

```
Github ID, Team Name
pconrad, Instructors
scottpchow, TAs
probablyfaiz, Students
phtcon, Instructors
scottpchow, Students
```

{ instructors: [pconrad, phtcon...], TAs: {...} }

### Step 2

Create those teams, Instructors, TAs, Students now exist on GitHub

### Step 3

Invite github users to teams via GitHub API

### Step 4

Run existing ``Refresh Github Teams`` job to associate students with the GitHub teams

### PR DESCRIPTION
**Pull request by ProbablyFaiz**
Full disclosure: this is messy and probably not that useful.

This PR implements the functionality described in #194 (and, as such, closes #194). It does the following things:

- Refactored job code to use an options hash in perform method to allow for any number of parameters to be passed to a job. No functionality changes here, but that's what all those ``options`` variables are.
- Minor refactor of github_machine_user into ``ApplicationController`` so the machine user code isn't repeated in a bunch of controllers.
- Adds the GitHub v3 API rate limit to admin dashboard. The v4 API does not support creating teams and editing their membership, so the new stuff relies on v3.
- A new view is added at endpoint ``/courses/:course_id/teams/create_teams`` and can be accessed via the "Manage Teams" menu on the teams show page of a course:
Screen Shot 2020-03-31 at 10 04 43 PM
Screen Shot 2020-03-31 at 10 07 13 PM
Your reaction may be that this view is kind of ugly. I agree.

- Upon uploading and submitting a CSV, a new job is triggered: ``CreateGithubTeamsJob``. It is passed a hash created from the CSV in which each key is a team name and each value is an array of its members. The central purpose of this job is to somehow not crash in the face of all the bullshit Github throws at it as it tries to create teams and add members to them. Here's why it kinda sucks:
    -  The big issue lies in the fact that when you try to create a team that already exists, instead of some useful response indicating that the team already exists, Octokit throws an exception. The same applies for adding a user to a team they're already in. So this job turns into a bit of begin/rescue spaghetti to work around that and do its best to create teams and populate them even if they already existed.
    - A secondary issue is that GitHub's endpoint for adding users to a team is case sensitive with regards to the username. I left a warning on the upload page, but it's annoying and will inevitably cause issues anyway.
    - Finally, because this functionality is not included in the GraphQL API, we're back to the stone age of one-by-one REST API calls which sucks for performance and conciseness reasons.

In conclusion, the GitHub API's team functionality is pretty anemic and this functionality isn't able to shroud much of that away. This feature works for the time being, but probably a smarter team creation functionality that involves students/emails rather than directly referencing GitHub IDs is much easier to use, though it would require a lot more implementation work.

### CODE REVIEW

**Comment by chundhau** [comment]
So this is complete? Could I please see a demo? @ProbablyFaiz @pconrad

**Comment by ProbablyFaiz** [comment]
@chundhau This was the page I briefly showed that takes in the ``username,team`` csv. We can show it in action next Monday.
